### PR TITLE
qemu: nios2: Enable UART 16550 driver for QEMU

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Enable UART 16550 driver for Nios-II QEMU platform.

Note: This PR is tested with patched version Qemu 3.0.0 which
adds support for altera_10m50_zephyr machine type.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

**Comment by nashif** [comment]
> Note: This PR is tested with patched version Qemu 3.0.0 which
> adds support for altera_10m50_zephyr machine type.

you have the patch?

File: dts/nios2/nios2-qemu.dtsi [hunk 1]
```diff
@@ -31,10 +31,10 @@
 		compatible = "simple-bus";
 		ranges;
 
-		uart0: uart@201000 {
```
Comment (nashif) [inline dts/nios2/nios2-qemu.dtsi]: cant we have both enabled? This way we can merge and start testing with new qemu and do the switch back and forth if needed. I would leave jtag until qemu is available.

**Review by nashif — CHANGES_REQUESTED** [review]

File: dts/nios2/nios2-qemu.dtsi (hunk 1 above)
Comment (ramakrishnapallala) [reply dts/nios2/nios2-qemu.dtsi]: Ok will keep the jtag option as well

**Comment by ramakrishnapallala** [comment]
"
Note: This PR is tested with patched version Qemu 3.0.0 which
adds support for altera_10m50_zephyr machine type.

you have the patch?
"

Yes Anas, i have the patch, I can attach it in the bug.

**Comment by ramakrishnapallala** [comment]
@nashif now the PR has support for both Jtag and NS16550 uarts and by default, we have selected jtag uart which is working with current SDK.

If we need to select ns16550, we have to change the following line in boards/nios2/qemu_nios2/qemu_nios2.dts:

- zephyr,console = &jtag_uart;
+ zephyr,console = &ns16550_uart;

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
rebase please @ramakrishnapallala

**Pull request merged**